firedrake/plot.py

```diff
@@ -862,7 +862,7 @@ def __call__(self, function):
         fiat_element = Q.finat_element.fiat_equivalent
         elem = fiat_element.tabulate(0, self._reference_points)[keys[dimension]]
         cell_node_list = Q.cell_node_list
-        data = function.dat.data_ro_with_halos[cell_node_list]
+        data = function.dat.data_ro_with_halos[cell_node_list].real
```

Comment:
This is the minimal thing, but should we adopt the same `complex_component` idea as used elsewhere in this file and have:

```python
def __call__(self, function, complex_component="real"):
    ...
    data = toreal(function.dat.data_ro_with_halos[cell_node_list], complex_component)
```
To allow people to plot the imaginary part.

Reply:
You're right. I've done essentially that, though the minimal change version.

Reply:
Ah, I hadn't spotted that `FunctionPlotter` is effectively an internal thing.